test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -102,16 +101,28 @@ void checkBlocks(TestBlock const& _blockFromFields, TestBlock const& _blockFromR
 
 void doBlockchainTests(json_spirit::mValue& _v, bool _fillin)
 {
-	if (!Options::get().fillchain) //fill blockchain through state tests
+	bool isRunFromStateTests = true;
+	string suitename = boost::unit_test::framework::get<boost::unit_test::test_suite>(boost::unit_test::framework::current_test_case().p_parent_id).p_name;
+	if (suitename.find("StateTests") == string::npos)
```

Comment:
Is there any way to pass the flag/enum to this function telling that we're in a special case of StateTests instead of querying suite name from boost? Why is `doBlockchainTests` called from StateTests suite anyway?